db/indexes.go

```diff
@@ -330,30 +300,29 @@ func (i *SGIndex) createIfNeeded(ctx context.Context, bucket base.N1QLStore, opt
 
 	// start a retry loop to create index,
 	worker := func() (shouldRetry bool, err error, value interface{}) {
-		err = bucket.CreateIndex(ctx, indexName, indexExpression, filterExpression, n1qlOptions)
+		err = bucket.CreateIndexIfNotExists(ctx, indexName, indexExpression, filterExpression, n1qlOptions)
 		if err != nil {
-			// If index has already been created (race w/ other SG node), return without error
-			if err == base.ErrAlreadyExists {
-				isDeferred = false // Index already exists, don't need to update.
-				return false, nil, nil
-			}
 			if strings.Contains(err.Error(), "not enough indexer nodes") {
 				return false, fmt.Errorf("Unable to create indexes with the specified number of replicas (%d).  Increase the number of index nodes, or modify 'num_index_replicas' in your Sync Gateway database config.", options.NumReplicas), nil
+			} else if errors.Is(err, base.ErrIndexBackgroundRetry) {
```

Comment:
Maybe drop a comment that we only think this will happen with <7.1 ?

Reply:
This error can happen on all versions. It's not related to the `IF NOT EXISTS` change, and has been added to change from per-index wait to collection-wide `waitForIndexesOnline`